src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -498,8 +541,8 @@ def _visit_max_over(self, node: foast.Call, **kwargs) -> foast.Call:
 
     def _visit_where(self, node: foast.Call, **kwargs) -> foast.Call:
```

Comment:
Actually, I just realized there is no test for any of this in `test_type_deduction`. 

The problem with this is that the types may be deduced completely wrongly here while all the tests still pass. Let's fix that before merging.